cloud-operations/packer-image-builder/main.tf

```diff
@@ -0,0 +1,129 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+locals {
+  compute_subnet_name = "image-builder"
+  compute_zone        = "${var.region}-a"
+}
+
+module "project" {
+  source          = "../../modules/project"
+  name            = var.project_id
+  parent          = try(var.root_node, null)
+  billing_account = try(var.billing_account, null)
+  project_create  = var.project_create
+  services = [
+    "compute.googleapis.com"
+  ]
+  service_config = {
+    disable_on_destroy         = false
+    disable_dependent_services = false
+  }
+}
+
+module "service-account-image-builder" {
+  source     = "../../modules/iam-service-account"
+  project_id = module.project.project_id
+  name       = "image-builder"
+  iam_project_roles = {
+    (var.project_id) = [
+      "roles/compute.instanceAdmin.v1",
+      "roles/iam.serviceAccountUser"
+    ]
+  }
+}
+
+module "service-account-image-builder-vm" {
+  source     = "../../modules/iam-service-account"
+  project_id = module.project.project_id
+  name       = "image-builder-vm"
+}
```

Comment:
Are there situations where the VM needs to log to Stackdriver, and its service account needs logging writer permissions? (just wondering, maybe there aren't)

Reply:
I would say that it highly depends on a particular case. In general it is not needed to build an image so I decided to leave it for further adoptions. 
Packer itself generates quite extensive output that should be logged, but that out of VM's scope.